### ISSUE TITLE
Sorting on timestamp columns doesn't work on Add Filter and Datatables

### ISSUE THREAD

**Issue by jqnatividad** [ckan/ckan#4298]
### CKAN Version if known (or site URL)
2.7.3

### Please describe the expected behaviour
When 1) creating a filter or 2) datatable sorting a timestamp column as declared in the associated resource data dictionary, you expect 1) to get a sorted timestamp list on the filter, or 2) the datatable is sorted by date order.

### Please describe the actual behaviour
For 1), you get a spinner that never returns.  For 2), the datatable, you get a popup with the error:

`DataTables warning: table id=dtprv - Ajax error. For more information about this error, please see http://datatables.net/tn/7`

Looking at the error log, you get:

``[Thu Jun 21 18:10:19.514014 2018] [:error] [pid 15201:tid 140389726643968] [remote 127.0.0.1:24846] Error - <class 'ckan.logic.ValidationError'>: {'info': {'params': [()], 'statement': ['\\n            SELECT array_to_json(array_agg(j))::text FROM (\\n                SELECT  "_id" as "_id", "STATION_ID" as "STATION_ID", "STATION_NAME" as "STATION_NAME", "FULL_STATION_NAME" as "FULL_STATION_NAME", "STATION_NUMBER" as "STATION_NUMBER", "STATION_TYPE" as "STATION_TYPE", "LATITUDE" as "LATITUDE", "LONGITUDE" as "LONGITUDE", "COUNTY_NAME" as "COUNTY_NAME", "SAMPLE_COUNT" as "SAMPLE_COUNT", to_json(to_char("SAMPLE_DATE_MIN", \\'YYYY-MM-DD"T"HH24:MI:SS\\')) as "SAMPLE_DATE_MIN", to_json(to_char("SAMPLE_DATE_MAX", \\'YYYY-MM-DD"T"HH24:MI:SS\\')) as "SAMPLE_DATE_MAX"\\n                FROM "24fc759a-ff0b-479a-a72a-c91a9384540f" \\n                 ORDER BY "SAMPLE_DATE_MIN" asc LIMIT 10 OFFSET 0\\n            ) AS j'], 'orig': ['could not identify an ordering operator for type json\\nLINE 5:                  ORDER BY "SAMPLE_DATE_MIN" asc LIMIT 10 OFF...\\n                                  ^\\nHINT:  Use an explicit ordering operator or modify the query.\\n']}, 'query': ['Invalid query']}
``

Here's a beautified version of the invalid SQL query:

```SQL
SELECT
	array_to_json (
	ARRAY_AGG ( j )) :: TEXT 
FROM
	(
	SELECT
		"_id" AS "_id",
		"STATION_ID" AS "STATION_ID",
		"STATION_NAME" AS "STATION_NAME",
		"FULL_STATION_NAME" AS "FULL_STATION_NAME",
		"STATION_NUMBER" AS "STATION_NUMBER",
		"STATION_TYPE" AS "STATION_TYPE",
		"LATITUDE" AS "LATITUDE",
		"LONGITUDE" AS "LONGITUDE",
		"COUNTY_NAME" AS "COUNTY_NAME",
		"SAMPLE_COUNT" AS "SAMPLE_COUNT",
		to_json (
		to_char( "SAMPLE_DATE_MIN", ' YYYY - MM - DD "T" HH24 : MI : SS ' )) AS "SAMPLE_DATE_MIN",
		to_json (
		to_char( "SAMPLE_DATE_MAX", ' YYYY - MM - DD "T" HH24 : MI : SS ' )) AS "SAMPLE_DATE_MAX" 
	FROM
		"24fc759a-ff0b-479a-a72a-c91a9384540f" 
	ORDER BY
		"SAMPLE_DATE_MIN" ASC 
	LIMIT 10 OFFSET 0 
	) AS j
```

### What steps can be taken to reproduce the issue? 

Upload a CSV with a timestamp column.  After upload, declare the column a timestamp, and then try to 1) add a filter on the timestamp column and 2) sort the column on the datatable.

**Comment by davidread** [comment]
This is very similar to #4288

**Comment by jqnatividad** [comment]
Managed to triage this issue, and get it to work by commenting out the to_json for timestamps

https://github.com/ckan/ckan/blob/master/ckanext/datastore/backend/postgres.py#L1683-L1684

**Comment by wardi** [comment]
I am working on this on #4288, @jqnatividad just for reference which version of postgres is this on?

**Comment by wardi** [comment]
fixed by #4288

**Comment by jqnatividad** [comment]
We're using Postgres 9.5.12 on RDS.  FWIW, we plan to move to AWS Aurora with the Postgres 9.6.6 engine next month.
